lib/segment/src/index/field_index/geo_index/mmap_geo_index.rs

```diff
@@ -0,0 +1,338 @@
+
+        let config: MmapGeoMapIndexConfig = read_json(&config_path)?;
+        let counts_per_hash =
+            unsafe { MmapSlice::try_from(mmap_ops::open_write_mmap(&counts_per_hash_path)?)? };
+        let points_map =
+            unsafe { MmapSlice::try_from(mmap_ops::open_write_mmap(&points_map_path)?)? };
+        let points_map_ids =
+            unsafe { MmapSlice::try_from(mmap_ops::open_write_mmap(&points_map_ids_path)?)? };
+        let point_to_values = MmapPointToValues::open(path)?;
+
+        let deleted = mmap_ops::open_write_mmap(&deleted_path)?;
+        let deleted = MmapBitSlice::from(deleted, 0);
+        let deleted_count = deleted.count_ones();
+
+        Ok(Self {
+            path: path.to_owned(),
+            counts_per_hash,
+            points_map,
+            points_map_ids,
+            point_to_values,
+            deleted: MmapBitSliceBufferedUpdateWrapper::new(deleted),
+            deleted_count,
+            points_values_count: config.points_values_count,
+            max_values_per_point: config.max_values_per_point,
+        })
+    }
+
+    pub fn check_values_any(
+        &self,
+        idx: PointOffsetType,
+        check_fn: impl Fn(&GeoPoint) -> bool,
+    ) -> bool {
+        self.deleted
+            .get(idx as usize)
+            .filter(|b| !b)
+            .map(|_| self.point_to_values.check_values_any(idx, |v| check_fn(&v)))
+            .unwrap_or(false)
+    }
+
+    pub fn values_count(&self, idx: PointOffsetType) -> usize {
+        self.deleted
+            .get(idx as usize)
+            .filter(|b| !b)
+            .and_then(|_| self.point_to_values.get_values_count(idx))
+            .unwrap_or(0)
+    }
+
+    pub fn get_points_per_hash(&self) -> impl Iterator<Item = (GeoHashRef, usize)> {
+        self.counts_per_hash
+            .iter()
+            .filter_map(|counts| Some((from_mmap_hash(&counts.hash)?, counts.points as usize)))
+    }
+
+    pub fn get_points_of_hash(&self, hash: &GeoHash) -> usize {
+        let hash = into_mmap_hash(hash);
+        if let Ok(index) = self
+            .counts_per_hash
+            .binary_search_by(|x| mmap_geo_hash_cmp(&x.hash, &hash))
+        {
+            self.counts_per_hash[index].points as usize
+        } else {
+            0
+        }
+    }
+
+    pub fn get_values_of_hash(&self, hash: &GeoHash) -> usize {
+        let hash = into_mmap_hash(hash);
+        if let Ok(index) = self
+            .counts_per_hash
+            .binary_search_by(|x| mmap_geo_hash_cmp(&x.hash, &hash))
+        {
+            self.counts_per_hash[index].values as usize
+        } else {
+            0
+        }
+    }
+
+    pub fn clear(self) -> OperationResult<()> {
+        let files = self.files();
+        let Self { path, .. } = self;
+        for file in files {
+            std::fs::remove_file(file)?;
+        }
+        let _ = remove_dir(path);
+        Ok(())
+    }
+
+    pub fn files(&self) -> Vec<PathBuf> {
+        todo!()
```

Comment:
Let's implement this

Reply:
Ahaha, sorry ) I Added missed implementation, there are no any another todos in `mmap_geo_index.rs`